sklearn/linear_model/tests/test_logistic.py

```diff
@@ -1824,3 +1853,40 @@ def test_scores_attribute_layout_elasticnet():
 
             avg_score_lr = cross_val_score(lr, X, y, cv=cv).mean()
             assert avg_scores_lrcv[i, j] == pytest.approx(avg_score_lr)
+
+
+def test_illconditioned_lbfgs():
+    # check that lbfgs converges even with ill-conditioned X
+    X, y = make_classification(n_samples=100, n_features=60, random_state=0)
+    X[:, 1] += 10000
+    X[:, 0] *= 10000
+    lr_pre = LogisticRegression(random_state=0, precondition=True)
+    with pytest.warns(None) as record:
+        lr_pre.fit(X, y)
+    assert len(record) == 0
+    loss_pre = _logistic_loss(
+        np.hstack([lr_pre.coef_.ravel(), lr_pre.intercept_]),
+        X, 2 * y - 1, 1)
+
+    lr = LogisticRegression(random_state=0, precondition=False)
+    with pytest.warns(ConvergenceWarning):
+        lr.fit(X, y)
+    loss = _logistic_loss(np.hstack([lr.coef_.ravel(), lr.intercept_]),
+                          X, 2 * y - 1, 1)
+    assert loss_pre < loss
```

Comment:
If I am not mistaken, this assertion is not guaranteed to hold because it does not take regularization into account. I think you should compare `lr.loss_values_` in this test instead.

Reply:
``_logistic_loss`` includes the regularizer. The last parameter is ``alpha``

Reply:
Thanks I had missed that. Maybe passing `alpha=1.` would make the code self explanatory. Maybe even renaming `_logistic_loss` to `_penalized_logistic_loss`.